x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecuritySystemIndices.java

```diff
@@ -683,4 +696,139 @@ private XContentBuilder getTokenIndexMappings() {
         }
     }
 
+    private SystemIndexDescriptor getSecurityProfileIndexDescriptor() {
+        return SystemIndexDescriptor.builder()
+            .setIndexPattern(".security-profile-[0-9]+*")
+            .setPrimaryIndex(INTERNAL_SECURITY_PROFILE_INDEX_8)
+            .setDescription("Contains user profile documents")
+            .setMappings(getProfileIndexMappings())
+            .setSettings(getProfileIndexSettings())
+            .setAliasName(SECURITY_PROFILE_ALIAS)
+            .setIndexFormat(INTERNAL_PROFILE_INDEX_FORMAT)
+            .setVersionMetaKey(SECURITY_VERSION_STRING)
+            .setOrigin(SECURITY_ORIGIN)
+            .setThreadPools(ExecutorNames.CRITICAL_SYSTEM_INDEX_THREAD_POOLS)
+            .build();
+    }
+
+    private Settings getProfileIndexSettings() {
+        return Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS, "0-1")
+            .put(IndexMetadata.SETTING_PRIORITY, 1000)
+            .put("index.refresh_interval", "1s")
+            .put(IndexMetadata.INDEX_FORMAT_SETTING.getKey(), INTERNAL_PROFILE_INDEX_FORMAT)
+            .build();
+    }
+
+    private XContentBuilder getProfileIndexMappings() {
+        try {
+            final XContentBuilder builder = jsonBuilder();
+            builder.startObject();
+            {
+                builder.startObject("_meta");
+                builder.field(SECURITY_VERSION_STRING, Version.CURRENT.toString());
+                builder.endObject();
+
+                builder.field("dynamic", "strict");
+                builder.startObject("properties");
+                {
+                    builder.startObject("uid");
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject("redirect_uid");
+                    builder.field("type", "keyword");
+                    builder.endObject();
+
+                    builder.startObject("enabled");
+                    builder.field("type", "boolean");
+                    builder.endObject();
+
+                    builder.startObject("user");
```

Comment:
Assuming this is up for discussion, I would think `profile` is more suitable.

Reply:
My take is that `profile` means the whole document and this section is the `user` of the profile. So I prefer keeping it as is.

Reply:
Okay, makes sense, thanks.
Do  you think nesting all these under a `profile` object would be better? From past experience I learned that we might end up storing multiple object types in the same index.

Reply:
I'd say No mainly because I think we are moving away (or at least want to move away) from storing multiple different types of documents in a single index. I am aware that the main security index is an example of hosting different type of documents. But I don't think it is something we should model after today. In fact, we had previous discussions about moving API keys out of the main security index (https://github.com/elastic/elasticsearch/issues/69685). The current general indexing strategy also favours smaller and more concrete indices. I also didn't add a `doc_type` field to the mapping along this line of thinking.